### PR TITLE
Migrate router, store and mixin files of stock page to TypeScript

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Stock page was missing some migrated files
| Type?             | refacto
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Stock page should work as expected
| Possible impacts? | Stock page behavior

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25750)

### CODE REVIEW

**Review by atomiix — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

File: admin-dev/themes/new-theme/js/app/pages/stock/store/index.ts [hunk 1]
```diff
@@ -22,13 +22,15 @@
...
+/* eslint-disable no-param-reassign */
+
+import Vue, {PluginObject} from 'vue';
 import Vuex from 'vuex';
 import _ from 'lodash';
 import * as actions from './actions';
 import mutations from './mutations';
 
-Vue.use(Vuex);
+Vue.use(<PluginObject<any>><unknown>Vuex);
```
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/js/app/pages/stock/store/index.ts]: Could you add a little comment, about why you have to cast it in `unknown`  then ` <PluginObject<any>>`, it's not really obvious ^^

Comment (NeOMakinG) [reply admin-dev/themes/new-theme/js/app/pages/stock/store/index.ts]: Needed to update Vuex in order to have the correct definition file, it's no more needed :)

**Review by Progi1984 — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]
Unfortunately, PR needs to be rebased :(

**Review by atomiix — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @NeOMakinG,

It is ok :heavy_check_mark: 
I checked multishop
I tried to filter by

- status
- categories
- supplier

one by one and all filters 
image

Import & export => OK

Search a supplier => ok
I tried with RTL language => ok
Display products below low stock level first => ok
Bulk edit quantity => ok
image

https://watch.screencastify.com/v/IzR7liNHAzzJZD6srK61

Thanks!

**Pull request merged**